File: app/src/main/java/com/ndmsystems/coala/layers/arq/data/InMemoryData.java
package com.ndmsystems.coala.layers.arq.data;

import java.util.Arrays;

/**
 * Created by Владимир on 16.08.2017.
 */

public class InMemoryData implements IData {

    private byte[] bytes;

    public InMemoryData(byte[] bytes) {
        this.bytes = bytes;
    }

    @Override
    public byte[] get(int from, int to) {
        return Arrays.copyOfRange(bytes, from, to);
    }

    @Override
    public byte[] get() {
        return bytes;
    }

    @Override
    public void append(IData data) {
        byte[] oldBytes = bytes;
        bytes = new byte[bytes.length+data.size()];
        System.arraycopy(oldBytes, 0, bytes, 0, oldBytes.length);
        System.arraycopy(data.get(), 0, bytes, oldBytes.length, data.size());
    }

    @Override
    public int size() {
        return bytes.length;
    }
}

File: app/src/main/java/com/ndmsystems/coala/layers/arq/data/InMemoryData.kt
package com.ndmsystems.coala.layers.arq.data

/**
 * Created by Владимир on 16.08.2017.
 */
class InMemoryData(private var bytes: ByteArray) : IData {
    override fun get(from: Int, to: Int): ByteArray? {
        return bytes.copyOfRange(from, to)
    }

    override fun get(): ByteArray {
        return bytes
    }

    override fun append(data: IData) {
        val oldBytes = bytes
        bytes = ByteArray(bytes.size + data.size())
        System.arraycopy(oldBytes, 0, bytes, 0, oldBytes.size)
        System.arraycopy(data.get(), 0, bytes, oldBytes.size, data.size())
    }

    override fun size(): Int {
        return bytes.size
    }
}